### PR TITLE
feat(service-portal): Finance service v2 client

### PR DESCRIPTION
**Pull request by Valur**
https://www.notion.so/Fj-rm-l-n-sti-fasi-729d88be2f46489ba042cd722333d6f1

## What

- Add new client for v2 of finance api

## Why

Needed for new finance transactions periods overview

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/api/domains/finance/src/lib/models/annualStatusDocument.model.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export class FinanceDocumentData {
   document!: string
 }
 
-@ObjectType()
+@ObjectType('FinanceDocumentModel')
 export class FinanceDocumentModel {
   @Field(() => FinanceDocumentData)
   docment!: FinanceDocumentData
```
Comment (thorkellmani) [inline libs/api/domains/finance/src/lib/models/annualStatusDocument.model.ts:15]:
```suggestion
  document!: FinanceDocumentData
```

**Review by thorkellmani — CHANGES_REQUESTED** [review]

**Review by thorkellmani — APPROVED** [review]
LGTM

File: libs/api/domains/finance/src/lib/models/annualStatusDocument.model.ts (hunk 1 above)
Comment (Valur) [reply libs/api/domains/finance/src/lib/models/annualStatusDocument.model.ts:15]: Out of scope fyrir þessa pullu

**Review by thordurhhh — APPROVED** [review]

File: libs/service-portal/finance/project.json [hunk 2]
```diff
@@ -24,6 +24,13 @@
       "options": {
         "command": "yarn ts-node -P libs/localization/tsconfig.lib.json libs/localization/scripts/extract 'libs/service-portal/finance/src/{lib,components,screens}/**/*.{js,ts,tsx}'"
       }
+    },
+    "schemas/codegen": {
```
[the remark this answers is not in the document]
Comment (Valur) [reply libs/service-portal/finance/project.json]: Updated with new format

File: libs/clients/finance-v2/project.json
Comment (saevarma) [inline libs/clients/finance-v2/project.json:1]: Is this integration not based on an OpenAPI document? 😱

File: libs/clients/finance-v2/src/lib/FinanceClientV2Config.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  schema,
+  load: (env) => ({
+    xRoadServicePath: env.required(
+      'XROAD_FINANCES_V2_PATH',
+      'IS-DEV/GOV/10021/FJS-Public/financeServicesFJS_v2',
+    ),
+    tokenExchangeScope: env.optionalJSON('XROAD_FINANCES_SCOPE') ?? [
+      '@fjs.is/finance',
+      // TODO: Remove when fjs has migrated to the scope above.
+      'api_resource.scope',
```
Comment (saevarma) [inline libs/clients/finance-v2/src/lib/FinanceClientV2Config.ts:21]: FJS should have stopped using `api_resource.scope` so please remove it.

**Review by lodmfjord — APPROVED** [review]
LGTM

File: libs/clients/finance-v2/project.json
Comment (Valur) [reply libs/clients/finance-v2/project.json:1]: OpenApi imlemented

File: libs/clients/finance-v2/src/lib/FinanceClientV2Config.ts (hunk 3 above)
Comment (Valur) [reply libs/clients/finance-v2/src/lib/FinanceClientV2Config.ts]: Removed

File: libs/api/domains/finance/src/lib/api-domains-finance.resolver.ts [hunk 4]
```diff
@@ -233,7 +233,7 @@ export class FinanceResolver {
   @Query(() => AssessmentYears)
   @Audit()
   async getAssessmentYears(@CurrentUser() user: User) {
-    return await this.financeServiceV2.getAssessmentYears(user.nationalId, user)
+    return await this.financeServiceV2.getAssessmentYears(user)
```
Comment (saevarma) [inline libs/api/domains/finance/src/lib/api-domains-finance.resolver.ts:236]: ~ FYI `return await` is unnecessary unless used in a try/catch

File: libs/clients/finance-v2/src/lib/financeV2.provider.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import {
+  ConfigType,
+  IdsClientConfig,
+  LazyDuringDevScope,
+  XRoadConfig,
+} from '@island.is/nest/config'
+import { Configuration, DefaultApi } from '../../gen/fetch'
+import { FinanceClientV2Config } from './financeV2.config'
+
+export const HmsLoansApiProvider: Provider<DefaultApi> = {
```
Comment (saevarma) [inline libs/clients/finance-v2/src/lib/financeV2.provider.ts:12]: Copy-paste?

**Review by saevarma — APPROVED** [review]
Removed `automerge` so you can address my 2cents before merging.

**Pull request merged**